gatsby-node.js

```diff
@@ -1,14 +1,92 @@
+
 /**
  * Implement Gatsby's Node APIs in this file.
  *
  * See: https://www.gatsbyjs.com/docs/node-apis/
  */
-// You can delete this file if you're not using it
+const fetch = require('node-fetch').default;
+const camelcase = require('camelcase').default;
 
-/**
- * You can uncomment the following line to verify that
- * your plugin is being loaded in your site.
- *
- * See: https://www.gatsbyjs.com/docs/creating-a-local-plugin/#developing-a-local-plugin-that-is-outside-your-project
- */
-exports.onPreInit = () => console.log("Loaded gatsby-starter-plugin")
+// TEMP: disabling console log warning.
+/* eslint-disable no-console */
+exports.sourceNodes = async ({
+  actions,
+  createContentDigest,
+  createNodeId,
+  getNodesByType
+}, options) => {
+  options = options || {};
+
+  if (!options.baseUrl) {
+    throw Error('You are missing the `baseUrl` option for gatsby-source-apostrophe');
+  }
+  if (!options.apiKey) {
+    throw Error('You are missing the `apiKey` option for gatsby-source-apostrophe');
+  }
+
+  if (Array.isArray(options.pieceTypes)) {
+    const routeBase = options.routeBase || 'api/v1';
+    for (const type of options.pieceTypes) {
+      const route = `${options.baseUrl}/${routeBase}/${type}`;
+      const response = await fetch(`${route}?apikey=${options.apiKey}`);
+      const data = await response.json();
+      const totalPages = data.pages;
+
+      const nodeTypeName = isAposCoreType(type)
+        ? convertCoreType(type)
+        : `Apos${camelcase(type, { pascalCase: true })}`;
+
+      await generatePieceNodes(data.results, nodeTypeName);
+
+      if (totalPages <= 1) {
+        continue;
+      };
+
+      for (let i = 2; i <= totalPages; i++) {
+        const pageResults = await getPiecePage(route, {
+          apiKey: options.apiKey,
+          page: i
+        });
+        await generatePieceNodes(pageResults, nodeTypeName);
+      }
+
+    }
+  }
+
+  async function generatePieceNodes (pieces, nodeType) {
+    const { createNode } = actions;
+    // loop through data and create Gatsby nodes
+    pieces.forEach(piece => {
+      piece.aposId = piece._id;
+      piece._id = undefined;
+      return createNode({
```

Comment:
What is this return statement achieving? No one is listening for a return value here. Is there a side effect on createNode that causes actual gatsby nodes to be stored somewhere?

Reply:
True. Lazy interpretation of their documentation.